invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/ResultsTableBody.js

```diff
@@ -1,40 +1,30 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { Button, Table } from 'semantic-ui-react';
+import { Table } from 'semantic-ui-react';
+import _get from 'lodash/get';
+import _isFunction from 'lodash/isFunction';
 
 export default class ResultsTableBody extends Component {
-  renderCell = (cell, column, id, colIndex) => {
-    return (
-      <Table.Cell key={colIndex + '-' + id} data-test={column + '-' + id}>
-        {cell}
-      </Table.Cell>
-    );
+  renderCell = (col, row, rowIndex) => {
+    if ('formatter' in col && _isFunction(col.formatter)) {
+      return col.formatter({ col: col, row: row, rowIndex: rowIndex });
+    }
+    return _get(row, col.field) || '-';
   };
 
   renderRow = (columns, rows) => {
-    return rows.map(row => {
-      const withRowAction = this.props.rowActionClickHandler ? (
-        <Button
-          compact
-          icon="info"
-          onClick={() => {
-            this.props.rowActionClickHandler(row);
-          }}
-          data-test={'btn-view-details-' + row.ID}
-        />
-      ) : null;
-
+    return rows.map((row, rowIndex) => {
+      const identifier = row.pid ? row.pid : row.id;
       return (
-        <Table.Row key={row.ID} data-test={row.ID}>
-          <Table.Cell textAlign="center">{withRowAction}</Table.Cell>
-          {columns.map((column, idx) =>
-            this.renderCell(
-              row[column] ? row[column] : '-',
-              column,
-              row.ID,
-              idx
-            )
-          )}
+        <Table.Row key={identifier} data-test={identifier}>
+          {columns.map(col => (
+            <Table.Cell
+              key={`${col.title}-${identifier}`}
+              data-test={`${col.title}-${identifier}`}
+            >
+              {this.renderCell(col, row, rowIndex)}
```

Comment:
wondering if you want to pass directly the value to be immediately usable...
```
{this.renderCell(_get(col.field, row), col, row, rowIndex)}
```

Reply:
I have tried this we can talk about it, it feels kind of magic to get it in the renderCell, and if formatter is present it will not even be used.